### PR TITLE
Add CLEAR MOT example

### PR DESCRIPTION
**Pull request by kopytjuk**
Since the calculation of CLEAR MOT metrics requires its own  `ClearMotAssociator`, I think it would be helpful to have an end-to-end example together with `MultiManager` where users just can copy&paste from. 

I built up on top of `06_DataAssociation-MultiTargetTutorial.py` tutorial to generate truths and tracks.

Since I am not a native speaker, feel free to correct my grammar and choice of words!

CC to @sdhiscocks, since he reviewed the other 3 PRs around CLEAR MOT.

### CODE REVIEW

**Comment by sdhiscocks** [comment]
Thanks @kopytjuk. Would it be possible to rebase this on the `main` branch? Unfortunately there was an issue (fixed this morning in 10b3ab7) that was stopping the documentation building on CI environment.

**Comment by kopytjuk** [comment]
> Thanks @kopytjuk. Would it be possible to rebase this on the `main` branch? Unfortunately there was an issue (fixed this morning in [10b3ab7](https://github.com/dstl/Stone-Soup/commit/10b3ab7acc325236cf42d6bbf4974b9d77d0c577)) that was stopping the documentation building on CI environment.

Done!

**Review by jswright-dstl — APPROVED** [review]
Great contribution. Just a few minor changes.

File: docs/examples/metrics/CLEAR_MOT.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+from stonesoup.plotter import AnimatedPlotterly
+
+plotter = AnimatedPlotterly(timesteps, tail_length=0.3)
+plotter.plot_ground_truths(truths, [0, 2])
+
+# %%
+# Generate detections with clutter
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# The detections (and clutter) from the truth tracks are later used as input for the
+# multiple-object tracking.
```
[the remark this answers is not in the document]
Comment (kopytjuk) [reply docs/examples/metrics/CLEAR_MOT.py]: Adapted

File: docs/examples/metrics/CLEAR_MOT.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+        truth_y = truth[k].state_vector[2]
+        for _ in range(np.random.randint(10)):
+            x = uniform.rvs(truth_x - 10, 20)
+            y = uniform.rvs(truth_y - 10, 20)
+            measurement_set.add(Clutter(np.array([[x], [y]]), timestamp=truth[k].timestamp,
+                                        measurement_model=measurement_model))
+    all_measurements.append(measurement_set)
+
+# %%
+# Run multiple-object-tracking
```
[the remark this answers is not in the document]
Comment (kopytjuk) [reply docs/examples/metrics/CLEAR_MOT.py]: Adapted

File: docs/examples/metrics/CLEAR_MOT.py [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+hypothesiser = DistanceHypothesiser(predictor, updater, measure=Mahalanobis(), missed_distance=3)
+
+
+from stonesoup.dataassociator.neighbour import GlobalNearestNeighbour
+
+data_associator = GlobalNearestNeighbour(hypothesiser)
+
+# %%
+# Run the Kalman filters.
+#
```
[the remark this answers is not in the document]
Comment (kopytjuk) [reply docs/examples/metrics/CLEAR_MOT.py]: Removed

File: docs/examples/metrics/CLEAR_MOT.py [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+    for track in tracks:
+        hypothesis = hypotheses[track]
+        if hypothesis.measurement:
+            post = updater.update(hypothesis)
+            track.append(post)
+        else:  # When data associator says no detections are good enough, we'll keep the prediction
+            track.append(hypothesis.prediction)
+
+# %%
+# Add tracks to the interacive plot
```
[the remark this answers is not in the document]
Comment (kopytjuk) [reply docs/examples/metrics/CLEAR_MOT.py]: Adapted

File: docs/examples/metrics/CLEAR_MOT.py [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+                                    tracks_key='tracks', truths_key='truths',
+                                    distance_measure=Euclidean((0, 2)))
+
+association_distance = 3.0  # meters
+clear_mot_associator = ClearMotAssociator(measure=Euclidean((0, 2)),
+                                          association_threshold=association_distance)
+metric_manager = MultiManager([clear_mot_metrics], associator=clear_mot_associator)
+
+# %%
+# add tracks data to metric manager
```
[the remark this answers is not in the document]
Comment (kopytjuk) [reply docs/examples/metrics/CLEAR_MOT.py]: Adapted

File: docs/examples/metrics/CLEAR_MOT.py [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+
+# %%
+# When associated, the average distance between tracks and truths is around 1m, which is reflected
+# by the MOTP metric.
+#
+# The MOTA score is around 0.57, which means that more than a half of the truth samples is
+# sucessfully tracked (i.e. below the association distance of 3 meters).
+# By oberving the proximity of truths and tracks, we can see that the green (south-east heading) is
+# followed by the violet track. At least half (50%) of the total number of truth samples
+# is successfully tracked.
```
[the remark this answers is not in the document]
Comment (kopytjuk) [reply docs/examples/metrics/CLEAR_MOT.py]: Adapted

File: docs/examples/metrics/CLEAR_MOT.py [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+            post = updater.update(hypothesis)
+            track.append(post)
+        else:  # When data associator says no detections are good enough, we'll keep the prediction
+            track.append(hypothesis.prediction)
+
+# %%
+# Add tracks to the interacive plot
+
+plotter.plot_tracks(tracks, [0, 2], uncertainty=False)
+
```
[the remark this answers is not in the document]
Comment (kopytjuk) [reply docs/examples/metrics/CLEAR_MOT.py:187]: Added

File: docs/examples/metrics/CLEAR_MOT.py [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+_ = truths.add(truth)
+
+# %%
+# Create an interactive plot instance and add the truth to it.
+
+from stonesoup.plotter import AnimatedPlotterly
+
+plotter = AnimatedPlotterly(timesteps, tail_length=0.3)
+plotter.plot_ground_truths(truths, [0, 2])
+
```
[the remark this answers is not in the document]
Comment (kopytjuk) [reply docs/examples/metrics/CLEAR_MOT.py:87]: Added

**Comment by kopytjuk** [comment]
@jswright-dstl thanks for your corrections, espcially those with spelling! I adopted all your suggestions.

File: docs/examples/metrics/CLEAR_MOT.py [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+# by the MOTP metric.
+#
+# The MOTA score is around 0.57, which means that more than a half of the truth samples are
+# successfully tracked (i.e. below the association distance of 3 metres).
+# By observing the proximity of truths and tracks, we can see that the green (south-east heading)
+# is followed by the violet track. At least half (i.e. 50%) of the total number of truth samples
+# is successfully tracked.
+#
+# While observing the red (north-east heading) truth, we see that it deviates from the
+# orange track at around the first third of the complete observation period. I.e. a third of the
```
Comment (csherman-dstl) [inline docs/examples/metrics/CLEAR_MOT.py:245]:
```suggestion
# While observing the red (north-east heading) truth, we see that the orange track deviates from
# it at around the first third of the complete observation period. I.e. a third of the
```

**Comment by kopytjuk** [comment]
@csherman-dstl @jswright-dstl hey guys, I just rebased to `main`, all tests seem to be green - after the rebase the `test-38` disappeared (was failing on my side). Feel free to merge :)

**Review by csherman-dstl — APPROVED** [review]

**Pull request merged**